wgpu-core/src/device/mod.rs

```diff
@@ -1294,6 +1294,11 @@ impl<G: GlobalIdentityHandlerFactory> Global<G> {
                     .expect("Failed to find binding declaration for binding");
                 let descriptor = match b.resource {
                     binding_model::BindingResource::Buffer(ref bb) => {
+                        if cfg!(debug_assertions) {
+                            // Asserts that the resource buffer exists.
+                            assert!(buffer_guard.contains(bb.buffer), "Buffer is invalid");
```

Comment:
would that be different from `debug_assert!`?

Reply:
Wasn't aware of `debug_assert!`, pretty neat 👍